src/pq/connection.cr

```diff
@@ -239,18 +239,21 @@ module PQ
           raise ConnectionError.new(
             "Only on is supported for integer_datetimes, got: #{frame.value.inspect}")
         end
-      else
-        # ignore
       end
+
+      true
     end
 
-    def connect
+    def connect(*, replication : String? = nil)
       startup_args = [
         "user", @conninfo.user,
         "database", @conninfo.database,
         "application_name", @conninfo.application_name,
         "client_encoding", "utf8",
       ]
+      if replication
+        startup_args << "replication" << replication
+      end
```

Comment:
`["replication", "true"]` enables physical-replication mode and `["replication", "database"]` enables logical-replication mode ([docs](https://www.postgresql.org/docs/current/protocol-replication.html)). `PG::Replication::Connection` [currently hardcodes it to `"database"`](https://github.com/will/crystal-pg/pull/301/files#diff-9db8c24b86773bc96353706cc4aec1b44c47b8ee0e06c8a2828d818b9605e004R19) to enable logical replication.